### PR TITLE
Add API to determines whether an object is present for the given key in the cache

### PR DESCRIPTION
**Pull request by maicki**
Adresses Issue #79

### CODE REVIEW

File: PINCache/PINCache.m [hunk 1]
```diff
@@ -294,6 +310,14 @@ - (NSUInteger)diskByteCount
     return byteCount;
 }
 
+- (BOOL)containsObjectForKey:(NSString *)key
+{
+    if (!key)
+        return NO;
```
Comment (jparise) [inline PINCache/PINCache.m]: Braces for consistency?

File: PINCache/PINCache.m [hunk 2]
```diff
@@ -74,6 +74,22 @@ + (instancetype)sharedCache
 
 #pragma mark - Public Asynchronous Methods -
 
+- (void)containsObjectForKey:(NSString *)key block:(PINCacheObjectContainmentBlock)block
+{
+    if (!key || !block) {
```
Comment (jparise) [inline PINCache/PINCache.m]: The `!block` early-return condition makes sense, but I'm not sure about `!key` not executing the block. It's such an exceptional case that I don't think it's that important in practice, but I also think it's a stronger contract to say that the block is always invoked if one is provided.

File: PINCache/PINDiskCache.m [hunk 3]
```diff
@@ -457,6 +457,22 @@ - (void)lockFileAccessWhileExecutingBlock:(void(^)(PINDiskCache *diskCache))bloc
     });
 }
 
+- (void)containsObjectForKey:(NSString *)key block:(PINDiskCacheContainsBlock)block
+{
+    __weak PINDiskCache *weakSelf = self;
+    
+    dispatch_async(_asyncQueue, ^{
+        PINDiskCache *strongSelf = weakSelf;
+        BOOL containsObject = [strongSelf containsObjectForKey:key];
```
Comment (jparise) [inline PINCache/PINDiskCache.m]: Move this call within the `if (block)` check? Otherwise, this is wasted work.

File: PINCache/PINDiskCache.m [hunk 4]
```diff
@@ -457,6 +457,22 @@ - (void)lockFileAccessWhileExecutingBlock:(void(^)(PINDiskCache *diskCache))bloc
...
+        
+        if (block) {
+            [strongSelf lock];
```
Comment (jparise) [inline PINCache/PINDiskCache.m]: Maybe I'm overlooking something, but why do we need to lock on reporting the result? Should the lock instead of around the `-containsObjectForKey:` call?

File: PINCache/PINMemoryCache.m [hunk 5]
```diff
@@ -270,6 +270,22 @@ - (void)trimToAgeLimitRecursively
 
 #pragma mark - Public Asynchronous Methods -
 
+- (void)containsObjectForKey:(NSString *)key block:(PINMemoryCacheContainmentBlock)block
+{
+    if (!key || !block) {
```
Comment (jparise) [inline PINCache/PINMemoryCache.m]: Similar comment as for the disk cache.

File: PINCache/PINCache.m (hunk 1 above)
Comment (maicki) [inline PINCache/PINCache.m]: We don't use the braces in other cases or do I overlook something?

Comment (jparise) [inline PINCache/PINCache.m]: You do above on Line 79-81.

Comment (maicki) [inline PINCache/PINCache.m]: Will change in line 79-81 and other places that I added braces to be consistent across the project.

File: PINCache/PINDiskCache.m (hunk 4 above)
Comment (maicki) [inline PINCache/PINDiskCache.m]: We actually don't need a lock in here as we already lock in `containsObjectForKey:`. Will change.

File: PINCache/PINCache.m (hunk 2 above)
Comment (maicki) [inline PINCache/PINCache.m]: If you look into the header file you can see that we declare key and block as `nonnull`. So currently the contract explicitly requires a key and block and the API consumer should not pass nil in one of the two arguments. We could discuss changing the block to `nullable`, but as the method does not have a side effect like calling `setObject:forKey:block:` it makes no really sense to call the method without a key or block argument. I decided to define both arguments `nonnull`. In this case we can early return if no key or block is given to prevent bad behavior if called from within Obj-C, Swift should already prevent API consumer to call it with a nil parameter in the first place.

**Comment by maicki** [comment]
I addressed @jparise comments. Can you take another look please. Thanks!

cc @jparise @garrettmoon

File: PINCache/PINCache.m (hunk 2 above)
Comment (jparise) [inline PINCache/PINCache.m]: Thanks, that makes sense. I missed those header annotations.

**Comment by jparise** [comment]
This looks good to me. Test coverage could be expanded a bit more to cover all of the edge cases, but that's probably true in general.

@garrettmoon for any final comments.

**Comment by garrettmoon** [comment]
Going to take a final look at this today, thanks @maicki and @jparise !

File: tests/PINCacheTests/PINCacheTests.m [hunk 6]
```diff
@@ -136,6 +136,27 @@ - (void)testObjectSetWithDuplicateKey
...
 
+- (void)testObjectContains
+{
+    NSString *key = @"key";
+    NSString *value = @"value";
+    
+    [self.cache setObject:value forKey:key];
+    
+    // Synchronously
+    XCTAssertTrue([self.cache containsObjectForKey:@"key"], @"object was gone");
```
Comment (garrettmoon) [inline tests/PINCacheTests/PINCacheTests.m]: nit: use key local var instead of `@"key"`

**Comment by garrettmoon** [comment]
Tests passed locally, so I'm merging. Dunno what's going on with Travis :/
